java/src/Ice/src/main/java/com/zeroc/Ice/ObjectAdapter.java

```diff
@@ -386,6 +388,13 @@ public interface ObjectAdapter {
    */
   Object findDefaultServant(String category);
 
+  /**
+   * Gets the dispatch pipeline of this object adapter.
```

Comment:
I find the logic around building the dispatch pipeline a little surprising.

This function makes it seem like it's a simple getter.
But in reality, calling this will cause us to build the dispatch-pipeline if one hasn't been built yet.
Which means you'll get an exception if you then try to add any extra middlewares.

At the least we should mention that "if the dispatch pipeline hasn't been built yet, we build-it."

Reply:
The logic is the same in all languages: the dispatch pipeline is built the first time you ask for it. This "first ask" is typically done by ConnectionI when it dispatches an incoming request.